### ISSUE TITLE
Procedural map - layers

### ISSUE THREAD

**Issue by DirectuX** [petrSchreiber/BetweenTheBubbles-tb#32]
Following up on Cornelius stress-test, and issue #29 

in order to avoid further strategic withdrawal, like from Durden to Cornelius, I would like to agree upon the essential gameZones the map has to hold.

As a start :
- _river_ gameZones vs _void_ gameZones
- lakes ( see #31 too)
- brooks

Do we need / want / keep: 
- gameZones properties ? 
  - depth of water
  - flow direction
  - flow strength
- waypoints
- banks gameZone
  - sand
  - grass
  - amphibian plants
  - dark soil
- locks
- unmoveable gameZone (may be difficult)

**Comment by petrSchreiber** [comment]
I think I need more detail on these parts, please (in any order, any time):
- locks - what are they?
- what is unmoveable gameZone?

I would say that there is not a single direction even within one game zone. I would say there would be one main current, but also some side ones, to make it harder/easier to reach goal in given gameZone.

**Comment by DirectuX** [comment]
> the essential gameZones **the map has to hold**

I'm not asking here of gameZones properties but what should be stored/created by the generated map.

locks : it can take any shape but as an example : a door that needs a key. If the Pilgrim brings the key when reaching the door, it opens to the "next level"

unmoveable gameZone #34  : already discussed upon the swirls : would have been needed to "connect" rivers of different grounds. We already agreed to be not necessary.

**Comment by petrSchreiber** [comment]
Superb, thanks a lot for the answers - I would propose to keep the state in the first post up to date, if I may, I would like to ask for marking unmoveable gameZone as ~~unmoveable GameZone~~ for example.

**Comment by DirectuX** [comment]
I thought to close the issue (a question) as soon as the question is answered, and open one or more new issues (focussed to do) with label accepted.

**Comment by petrSchreiber** [comment]
Sure, we can go this way, but I wonder what to do with the original question on the board then? Remove from board?

**Comment by DirectuX** [comment]
it will be  closed : it won't be displayed any more isn't it ?

**Comment by DirectuX** [comment]
Please look at the picture in #29.

As a start :

- river gameZones vs void gameZones (**illustrated**)
- lakes ( see #31 too) (**illustrated**)
- brooks (**illustrated**)

Do we need / want / keep:

- gameZones properties ?
  - depth of water
  - flow direction
  - flow strength
- waypoints
- banks gameZone (**illustrated**)
  - sand
  - grass
  - amphibian plants
  - dark soil
- locks (**illustrated**)
- ~~unmoveable gameZone (may be difficult)~~


What I think to keep was **illustrated**.
Please, what are your thoughts about waypoints and gameZones properties ?

**Comment by petrSchreiber** [comment]
Waypoints - I think there should be a current, which will flow strong through the main paths.

GameZone properties I put on my TODO list.

**Comment by DirectuX** [comment]
> Waypoints - I think there should be a current, which will flow strong through the main paths.

This comment and the ECS Poc make me think : the answer to the current may be without waypoints: each water gamezone can a have a _flowDirection_ property (%NORTH, %SOUTH, %EAST, %WEST) and a flow strength property. At map generation moment, we can store the value according to world's settings (a brook from a marsh ≠ a rapid, though they may have the same width) and according to the width of the water.

**Comment by petrSchreiber** [comment]
Let's get rid of waypoints - the proposed idea sounds good to me 👍

**Comment by petrSchreiber** [comment]
_moved_

**Comment by DirectuX** [comment]
I agree with these ideas, but the issue was about the gameZones data stored by the procedurally generated map.
This comment may have better place in #33 ?

**Comment by petrSchreiber** [comment]
Good point, moved 👍 Will think about this one again 👍
